qa/rpc-tests/test_framework/util.py

```diff
@@ -273,7 +300,11 @@ def initialize_chain(test_dir,bitcoinConfDict=None,wallets=None, bins=None):
             for peer in range(4):
                 for j in range(25):
                     set_node_times(rpcs, block_time)
-                    rpcs[peer].generate(1)
+                    try:
+                        rpcs[peer].generate(1)
+                    except Exception as e:
```

Comment:
Always catch the specific exception that is OK, especially in library code.

But in this case we want there to be an issue if the node supposed to be able to generate, but cannot.  What do you think about creating node capabilities member variables when the node objects are constructed, and then having something like:
if rpcs[peer].miningCapable:
    for j in range(25):

Reply:
Good idea.  Where would you like to add this capability. 
Is it to the C++ source code, or the python code.

If it is on the C++ side, I tried to find a node class in the source code. Which file contains this class?

If it is on the framework, the how can you access it from the rpcs[peer]
rpcs is a proxy to the rpc connection to the server handling the msg for the node.

Reply:
This capability variable is added to the C++ source?